### PR TITLE
Improve Package Detection 

### PR DESCRIPTION
**Pull request by jacobthesheep**
## Changes

- Update preferred-pm and which-pm-runs as latest version now supports Bun.
- Standardized variable names from `pkgManager` to `packageManager`.
- `bunx astro add` command correctly uses bun.
- Telemetry notice returns package manager.

### Previous attempts

https://github.com/withastro/astro/pull/8219
https://github.com/withastro/astro/pull/8150

## Testing

Manually tested. Here's a screenshot
okay
bun-telemetry

## Docs

https://docs.astro.build/en/recipes/bun/

### CODE REVIEW

**Comment by jacobthesheep** [comment]
preferred-pm also now supports bun but which-pm is still yet too

**Comment by jacobthesheep** [comment]
Waiting on https://github.com/zkochan/packages/pull/179 to be merged.

File: packages/astro/src/cli/index.ts [hunk 1]
```diff
@@ -127,8 +128,10 @@ async function runCommand(cmd: string, flags: yargs.Arguments) {
 		process.env.NODE_ENV = cmd === 'dev' ? 'development' : 'production';
 	}
 
+	const packageManager = (await whichPm(process.cwd())).name ?? 'npm';
+
 	const { notify } = await import('./telemetry/index.js');
-	await notify();
+	await notify(packageManager);
```
Comment (bluwy) [inline packages/astro/src/cli/index.ts:134]: Line 131 should be moved into the `notify()` function, we usually keep this file simple and only re-route commands to their main function.

File: .changeset/friendly-clocks-act.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+'create-astro': patch
+'@astrojs/telemetry': patch
```
Comment (bluwy) [inline .changeset/friendly-clocks-act.md:3]: I doesn't seem like the telemetry package is changed, I think we can remove this?

File: .changeset/friendly-clocks-act.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+'astro': patch
+---
+
+Update preferred-pm and which-pm-runs packages as the latest versions now support Bun.
+
+Standardized variable naming by changing all instances of pkgManager to packageManager to ensure consistent wording throughout the codebase.
+
+`bunx astro add` command correctly uses bun.
+
+Telemetry notice returns package manager.
```
Comment (bluwy) [inline .changeset/friendly-clocks-act.md:13]: I think we could keep this a bit simpler. The changelog here is usually user-facing so we can simplify it, e.g.
```suggestion
Support detecting Bun when logging messages with package manager information
```

File: .changeset/friendly-clocks-act.md (hunk 2 above)
Comment (jacobthesheep) [reply .changeset/friendly-clocks-act.md:3]: Okay.

**Review by bluwy — APPROVED** [review]

**Pull request merged**